cli/repair/common.go

```diff
@@ -0,0 +1,180 @@
+package repair
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/fatih/color"
+	"github.com/pmezard/go-difflib/difflib"
+	"github.com/tarantool/cartridge-cli/cli/context"
+	"github.com/tarantool/cartridge-cli/cli/project"
+)
+
+func getAppWorkDirNames(ctx *context.Ctx) ([]string, error) {
+	if err := project.SetSystemRunningPaths(ctx); err != nil {
+		return nil, fmt.Errorf("Failed to get default paths: %s", err)
+	}
+
+	if fileInfo, err := os.Stat(ctx.Running.DataDir); os.IsNotExist(err) {
+		return nil, fmt.Errorf("Data directory %s doesn't exist", ctx.Running.DataDir)
+	} else if err != nil {
+		return nil, fmt.Errorf("Failed to use specified data directory: %s", err)
+	} else if !fileInfo.IsDir() {
+		return nil, fmt.Errorf("%s is not a directory", ctx.Running.DataDir)
+	}
+
+	if ctx.Cli.Verbose {
+		log.Infof("Data directory is set to: %s", ctx.Running.DataDir)
+	}
+
+	workDirs, err := ioutil.ReadDir(ctx.Running.DataDir)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to list data directory: %s", err)
+	}
+
+	appWorkDirsPrefix := fmt.Sprintf("%s.", ctx.Project.Name)
+	appWorkDirNames := make([]string, 0)
+	for _, workDir := range workDirs {
+		workDirName := workDir.Name()
+		if strings.HasPrefix(workDirName, appWorkDirsPrefix) {
+			appWorkDirNames = append(appWorkDirNames, workDirName)
+		}
+	}
+
+	if len(appWorkDirNames) == 0 {
+		return nil, fmt.Errorf("No instance working directories found in %s", ctx.Running.DataDir)
+	}
+
+	return appWorkDirNames, nil
+}
+
+func getBackupPath(path string) string {
+	return fmt.Sprintf("%s.bak", path)
+}
+
+func createFileBackup(path string) error {
+	if _, err := os.Stat(path); err != nil {
+		return fmt.Errorf("Failed to use specified path: %s", err)
+	}
+
+	backupPath := getBackupPath(path)
+	if _, err := os.Stat(backupPath); err != nil {
+		os.RemoveAll(backupPath)
+	}
+
+	file, err := os.Open(path)
```

Comment:
Please specify Open flags explicitly

Reply:
Will be fixed in #339

Reply:
`os.Open` uses `RDONLY` flag by default. I've added file mode flags to created backup file.

See [commit](https://github.com/tarantool/cartridge-cli/pull/339/commits/4ccad749c279cf73dc702d142cf8a3d2e456f7f6)